modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2TreeInlineObjectDetector.java

```diff
@@ -68,26 +85,61 @@
                 continue;
```

Comment:
Can there be a problem if `Value.UNKNOWN` was inlined and `fullSize` returns some garbage?

Reply:
good point. However it will work as is. fullSize return in this case value greater or equals 3